### PR TITLE
Optimizing SetBuoyancy. Removed isnan. Added Ent:Get/SetFriction

### PR DESCRIPTION
**Pull request by dvdvideo1234**
Tested and working in one commit !

### CODE REVIEW

**Comment by AbigailBuccaneer** [comment]
Looks good to me. Any objections?

File: lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua [hunk 1]
```diff
@@ -21,5 +21,7 @@ E2Helper.Descriptions["propDrag"] = "Passing 0 makes the entity not be affected
 E2Helper.Descriptions["propDraw"] = "Passing 0 disables rendering for the entity (makes it really invisible)"
 E2Helper.Descriptions["propShadow"] = "Passing 0 disables rendering for the entity's shadow"
 E2Helper.Descriptions["propSetBuoyancy"] = "Sets the prop's buoyancy ratio from 0 to 1"
+E2Helper.Descriptions["propSetFriction"] = "Sets prop's friction coefficient ( normal is 1 )"
```
Comment (Divran) [inline lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua]: nitpicking, but it should be "(default is 1)"

**Comment by dvdvideo1234** [comment]
Done ! ;D

**Comment by arduinium** [comment]
```
local friction = math.Clamp(friction, -1000, 1000)
this:SetFriction(friction)
```
There is no need to create a variable if you use it only once. Just use
```
this:SetFriction( math.Clamp(friction, -1000, 1000) )
```

Same thing in propSetBuyoancy ;)

**Comment by dvdvideo1234** [comment]
I know but that way is more organized and read-friendly if you know what I men ;D

**Comment by dvdvideo1234** [comment]
There is no single word for a whole day... I guess the silence really means agreement ...

**Comment by AbigailBuccaneer** [comment]
I'll pull this when I get home from work unless anybody has any more comments in the next couple of hours.

**Pull request merged**